app/models/user.rb

```diff
@@ -1,13 +1,14 @@
 class User < ApplicationRecord
-  attr_accessor :remember_token
-  before_save{email.downcase!}
+  attr_accessor :remember_token, :activation_token
```

Comment:
dòng này có tác dụng gì vậy em?

Reply:
tìm hiểu thêm và trả lời chi tiết nữa nhé em

Reply:
attr_accessor sẽ tạo 2 biến instance @remember_token và @activation_token, sau đó nó sẽ thêm các phương thức đọc , ghi , để từ bên ngoài có thể gọi User.activation_token hoặc gán User.activation_token = ạ.

Reply:
attr_accessor is just a method, what it does is creating 2 instance variables @remember_token, @activation_token and it also creating these methods: remember_token, remember_token=, activation_token, activation_token= for read and write those instance variables.
(if you just want to read/write you can use attr_reader/attr_writer)